src/grpc/infer_handler.cc

```diff
@@ -497,10 +498,7 @@ InferGRPCToInput(
             auto& serialized = serialized_data->back();
             serialized.reserve(
                 io.contents().int_contents_size() * elem_byte_size);
-            for (const auto& element : io.contents().int_contents()) {
-              // Assuming the system is little-endian, picking the
-              // least significant byte of 32-bit integer as a
-              // int8 element
+            for (const int8_t element : io.contents().int_contents()) {
```

Comment:
small nit: can we use "const auto"? Are we not using reference because the reference is larger than the actual data so making a copy of reference is slower than actually copying the data?

Reply:
const auto  will use `int32_t` and then the append below only works correctly for little-endian (it copies the first byte of the 4 byte `int32_t`).

coercing to int8_t forces the compiler to cast to int8_t correctly for the native architecture and then we can copy it independent of big or little endian (i.e. it gets the correct byte either first or last). 

This seemed like the cleanest / most efficient way to handle this (essentially remove the assumption on the little endian byte order)

Reply:
I learned something new today 👍

Reply:
Thanks for explaining! Resolving.